### PR TITLE
Split up tools/run_tests/helper_scripts/run_ruby_end2end_tests.sh into granular targets

### PR DESCRIPTION
**Pull request by apolcyn**
For better reporting and more precise failure history

This also renames the affected test script "main" files to be suffixed with `_test.rb`.

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -926,11 +926,31 @@ def test_specs(self):
...
+                'src/ruby/end2end/errors_load_before_grpc_lib_test.rb',
+                'src/ruby/end2end/logger_load_before_grpc_lib_test.rb',
+                'src/ruby/end2end/status_codes_load_before_grpc_lib_test.rb',
+                'src/ruby/end2end/call_credentials_timeout_test.rb',
+                'src/ruby/end2end/call_credentials_returning_bad_metadata_doesnt_kill_background_thread_test.rb'
+        ]:
+            tests.append(
+                self.config.job_spec(['ruby', test],
+                                     shortname=test,
+                                     timeout_seconds=20 * 60,
```
Comment (jtattermusch) [inline tools/run_tests/run_tests.py:952]: 20 mins timeout for each of the tests ran separately sounds like a bit too much.

File: tools/run_tests/run_tests.py [hunk 2]
```diff
@@ -926,11 +926,31 @@ def test_specs(self):
                                  timeout_seconds=10 * 60,
                                  environ=_FORCE_ENVIRON_FOR_WRAPPERS)
         ]
-        tests.append(
-            self.config.job_spec(
-                ['tools/run_tests/helper_scripts/run_ruby_end2end_tests.sh'],
-                timeout_seconds=20 * 60,
-                environ=_FORCE_ENVIRON_FOR_WRAPPERS))
+        for test in [
+                'src/ruby/end2end/sig_handling_test.rb',
```
Comment (jtattermusch) [inline tools/run_tests/run_tests.py:930]: is there a way to make sure that when new tests are added, we won't forget including them here?

Comment (apolcyn) [reply tools/run_tests/run_tests.py:930]: There isn't. But there wasn't a way to be sure this this previously when test cases were listed in the shell script, either.

One thing we could do which could perhaps clean this up and avoid listing test cases here is for `run_tests.py` to run every ruby script under `src/ruby/end2end` that ends with `_test.rb`.

Any thoughts?

Comment (jtattermusch) [reply tools/run_tests/run_tests.py:930]: What I ended up doing in C# is to add an extra C# test ("c# sanity test") that only does one thing - lists all the existing C# tests and makes sure that matches https://github.com/grpc/grpc/blob/master/src/csharp/tests.json (which is where run_tests.py gets the list of C# tests from).
https://github.com/grpc/grpc/blob/fc4c9d74eebfdfbf99f181ebfd377f6a01db2946/src/csharp/Grpc.Core.Tests/SanityTest.cs#L44

For ruby you could do something similar - have a ruby tests.json, use it in run_tests.py and also check its contents by a dedicated ruby test (which reads the .json and also lists all the *_test.rb files and compares the two sets).

**Pull request merged**